de.tu_bs.cs.isf.e4cf.parts.project_explorer/src/de/tu_bs/cs/isf/e4cf/parts/project_explorer/controller/ProjectExplorerToolBarController.java

```diff
@@ -0,0 +1,194 @@
+package de.tu_bs.cs.isf.e4cf.parts.project_explorer.controller;
+
+
+import org.eclipse.swt.widgets.Shell;
+
+import de.tu_bs.cs.isf.e4cf.core.stringtable.E4CEventTable;
+import de.tu_bs.cs.isf.e4cf.core.util.ServiceContainer;
+import de.tu_bs.cs.isf.e4cf.parts.project_explorer.handlers.FileImportHandler;
+import de.tu_bs.cs.isf.e4cf.parts.project_explorer.handlers.NewFolderHandler;
+import de.tu_bs.cs.isf.e4cf.parts.project_explorer.handlers.RemoveFileCommand;
+import de.tu_bs.cs.isf.e4cf.parts.project_explorer.handlers.ShowInExplorerHandler;
+import de.tu_bs.cs.isf.e4cf.parts.project_explorer.stringtable.FileTable;
+import javafx.event.EventHandler;
+import javafx.event.ActionEvent;
+import javafx.scene.control.Button;
+import javafx.scene.control.Separator;
+import javafx.scene.control.TextField;
+import javafx.scene.control.ToolBar;
+import javafx.scene.control.Tooltip;
+import javafx.scene.image.ImageView;
+
+/**
+ * Fills and handles the ProjectExplorer Toolbar.
+ * Please note that the use of FX Accelerators as shortcuts is discouraged as FX <13 consumes their KeyCombos application wide
+ * and this is not desired behavior.
+ */
+public class ProjectExplorerToolBarController {
+
+	private ServiceContainer services;
+
+	private ToolBar bar;
+
+	private Button btnUndo;
+	private Button btnRedo;
+	private Button btnNewFolder;
+	private Button btnImportFiles;
+	private Button btnDelete;
+	private Button btnShowExplorer;
+	boolean isFlatView = false;
+
+	/**
+	 * Fills the toolbar with Buttons performing project explorer related actions.
+	 * @param toolbar The FXML Toolbar Object defined in UI
+	 * @param services The ServiceContainer that holds all e4c Services, required for some handlers
+	 * @param shell The current UI Shell passed on from the ViewController's Parent, required for some handlers
+	 */
+	public ProjectExplorerToolBarController(ToolBar toolbar, ServiceContainer services, Shell shell) {
+		bar = toolbar;
+		this.services = services;
+
+		// Search Section
+		TextField search = new TextField();
+		search.setVisible(false);
+		search.setOnAction(actionEvent -> {
+			services.eventBroker.send(E4CEventTable.EVENT_FILTER_CHANGED, search.getText());
+		});
```

Comment:
Would be cool if the view updates in real time with every keystroke. But maybe thats too demanding to update all the time.

Reply:
As we need to update the tree every time I would hesitate for now.
I could imagine larger workspaces may be heavily impacted by such behaviour.

If no one else speaks out about it, I would postpone that and maybe make another issue for that feature.
Maybe including a switch to toggle the behaviour.